lambda_service/insert_data.py

```diff
@@ -56,36 +42,49 @@ def sendToSubscribers(data):
     dataToSend = {"messages": [data]}
     for connectionID in connections:
         try: 
-            connectionResponse = _send_to_connection2(gatewayApi, connectionID, dataToSend, os.getenv('WSS_URL'))
-            print('connection response: ')
-            print(json.dumps(connectionResponse))
+            connectionResponse = _send_to_connection(
+                    gatewayApi, connectionID, dataToSend, os.getenv('WSS_URL'))
+            logger.info((
+                f"connection response: "
+                f"{json.dumps(connectionResponse)}")
+            )
         except gatewayApi.exceptions.GoneException:
-            print(f"Failed sending to {connectionID} due to GoneException. Removing it from the connections table.")
+            error_msg = (
+                f"Failed sending to {connectionID} due to GoneException. "
+                "Removing it from the connections table."
+            )
+            logger.exception(error_msg)
             _remove_connection(connectionID)
             continue
 
-def main2(event, context):
-    sendToSubscribers("hello")
 
-def main(event, context):
+def handle_s3_object_created(event, context):
+
+    logger.info(event)
     
     # Get the object from the event and show its content type
     bucket = event['Records'][0]['s3']['bucket']['name']
     
-    # Sample file_path is like: WMD/raw_data/2019/WMD-ea03-20190621-00000007-EX00.fits.bz2
-    file_path = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'], encoding='utf-8')
+    # Sample file_path is like: data/wmd-ea03-20190621-00000007-EX00.fits.bz2
+    file_path = urllib.parse.unquote_plus(
+            event['Records'][0]['s3']['object']['key'], 
+            encoding='utf-8'
+        )
     
     # Format amazon upload timestamp (yyyy-mm-ddThh:mm:ss.mmmZ) 
     # to psql timestamp without timezone (yyyy-mm-dd hh:mm:ss).
-    last_modified = list(urllib.parse.unquote_plus(event['Records'][0]['eventTime']))
+    last_modified = list(urllib.parse.unquote_plus(
+            event['Records'][0]['eventTime']))
     last_modified[10] = ' '
     last_modified = last_modified[:-5]
     last_modified = "".join(last_modified)
     
-    # The 'filename' that looks somethign like 'WMD-ea03-20190621-00000007-EX00.fits.bz2'
+    # The 'filename' that looks something like 
+    # 'wmd-ea03-20190621-00000007-EX00.fits.bz2'
     file_key = file_path.split('/')[-1]
     
-    # The base_filename (aka primary key) is something like 'WMD-ea03-20190621-00000007'
+    # The base_filename (aka primary key) is something like 
+    # 'wmd-ea03-20190621-00000007'
     base_filename = file_key[:26]
```

Comment:
This seems pretty fragile. Is there a better way to parse the base filename rather than relying on indexing? Some kind of delimiter, maybe?

Reply:
I agree, though I'm not sure of a better solution if the filename had a different format. Do you think a delimiter (ie. splitting on '-') would be more reliable if the format did unexpectedly change?

Reply:
Yeah, that's tricky. If you're confident that the base filename will have `-EX...` after it, that seems safer? 

like a `file_key.split('EX')[0]` would be okay. Obviously it's not ideal, but as long as that `EX` doesn't creep its way into your base filename naming convention sometime in the future I think you'll be okay

Also this way, if something does unexpectedly change, you'll have a better chance of seeing errors. With the 26-character check something could change and it might not error out.

Reply:
Hmm, that's a great point about string slicing not catching errors. I guess it's just a tricky issue because it's not easy to guess what an incorrect input might be like (for example, the instrument name might be EX03, rendering the split('-EX') method problematic). Not to mention parsing the rest of the string up in a reliable way, like the site abbreviation from the first three characters. 

What do you think of keeping the same slicing methods, but also adding a filename verifier so that changes to the filename are still caught appropriately?

Reply:
Added verifier in the latest commit f70f218

Reply:
Cool, that makes sense to me